rust/processor/src/processors/parquet_processors/parquet_transaction_metadata_processor.rs

```diff
@@ -0,0 +1,150 @@
+
+pub struct ParquetTransactionMetadataProcessor {
+    connection_pool: ArcDbPool,
+    write_set_size_info_sender: AsyncSender<ParquetDataGeneric<WriteSetSize>>,
+}
+
+impl ParquetTransactionMetadataProcessor {
+    pub fn new(
+        connection_pool: ArcDbPool,
+        config: ParquetTransactionMetadataProcessorConfig,
+        new_gap_detector_sender: AsyncSender<ProcessingResult>,
+    ) -> Self {
+        if let Some(credentials) = config.google_application_credentials.clone() {
+            std::env::set_var(GOOGLE_APPLICATION_CREDENTIALS, credentials);
+        }
+
+        let write_set_size_info_sender = create_parquet_handler_loop::<WriteSetSize>(
+            new_gap_detector_sender.clone(),
+            ProcessorName::ParquetTransactionMetadataProcessor.into(),
+            config.bucket_name.clone(),
+            config.bucket_root.clone(),
+            config.parquet_handler_response_channel_size,
+            config.max_buffer_size,
+            config.parquet_upload_interval_in_secs(),
+        );
+        Self {
+            connection_pool,
+            write_set_size_info_sender,
+        }
+    }
+}
+
+impl Debug for ParquetTransactionMetadataProcessor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "ParquetTransactionMetadataProcessor {{ capacity of write set size info channel: {:?} }}",
+            self.write_set_size_info_sender.capacity(),
+        )
+    }
+}
+
+#[async_trait]
+impl ProcessorTrait for ParquetTransactionMetadataProcessor {
+    fn name(&self) -> &'static str {
+        ProcessorName::ParquetTransactionMetadataProcessor.into()
+    }
+
+    async fn process_transactions(
+        &self,
+        transactions: Vec<Transaction>,
+        start_version: u64,
+        end_version: u64,
+        _: Option<u64>,
+    ) -> anyhow::Result<ProcessingResult> {
+        let last_transaction_timestamp = transactions.last().unwrap().timestamp.clone();
+        let mut transaction_version_to_struct_count: AHashMap<i64, i64> = AHashMap::new();
+
+        let mut write_set_sizes = vec![];
+
+        for txn in &transactions {
+            let txn_version = txn.version as i64;
+            let block_timestamp = parse_timestamp(txn.timestamp.as_ref().unwrap(), txn_version);
+            let size_info = match txn.size_info.as_ref() {
+                Some(size_info) => size_info,
+                None => {
+                    warn!(version = txn.version, "Transaction size info not found");
+                    continue;
+                },
+            };
+            for (index, write_set_size_info) in size_info.write_op_size_info.iter().enumerate() {
+                write_set_sizes.push(WriteSetSize::from_transaction_info(
+                    write_set_size_info,
+                    txn_version,
+                    index as i64,
+                    block_timestamp,
+                ));
+                transaction_version_to_struct_count
```

Comment:
what will happen if `txn.size_info.as_ref()` is None? dose gap detector can handle this? if not, we should upgrade to panic or something

Reply:
if that is None, wouldn't that proceed to next txn?

Reply:
yea i was thinking there were some consecutive constrain required by Gap detector and now gap detector would look like [v1, v2, v4, v5..], thus the question

Reply:
no, the current behavior is, if there is no structs to process (e.g. the v3 will have 0 count),  it will proceed to next one to process. b/c we loop through starting_version ..endIng_version to populate the map first aside from the result we handle here.